test/expected/alter.out

```diff
@@ -194,3 +194,136 @@ SELECT * FROM _timescaledb_catalog.chunk WHERE id = 2;
+    tablename     | tablespace 
+------------------+------------
+ _hyper_1_1_chunk | 
+ _hyper_2_3_chunk | 
+ _hyper_2_4_chunk | 
+ _hyper_3_5_chunk | new_space
+ _hyper_3_6_chunk | new_space
+ _hyper_3_7_chunk | new_space
+ _hyper_3_8_chunk | 
+ hyper_in_space   | new_space
+(8 rows)
+
+SET default_tablespace = new_space;
+-- should be inserted in the new_space tablespace which is now default
+INSERT INTO hyper_in_space(time, temp, device) VALUES (11, 24, 3);
+SELECT tablename, tablespace FROM pg_tables
+WHERE tablename = 'hyper_in_space' OR tablename LIKE '\_hyper\__\__\_chunk' ORDER BY tablename;
+    tablename     | tablespace 
+------------------+------------
+ _hyper_1_1_chunk | 
+ _hyper_2_3_chunk | 
+ _hyper_2_4_chunk | 
+ _hyper_3_5_chunk | new_space
+ _hyper_3_6_chunk | new_space
+ _hyper_3_7_chunk | new_space
+ _hyper_3_8_chunk | 
+ _hyper_3_9_chunk | new_space
+ hyper_in_space   | new_space
+(9 rows)
+
+-- the one chunk in pg_default should be moved over
+ALTER TABLE hyper_in_space SET TABLESPACE new_space;
+SELECT tablename, tablespace FROM pg_tables
+WHERE tablename = 'hyper_in_space' OR tablename LIKE '\_hyper\__\__\_chunk' ORDER BY tablename;
+    tablename     | tablespace 
+------------------+------------
+ _hyper_1_1_chunk | 
+ _hyper_2_3_chunk | 
+ _hyper_2_4_chunk | 
+ _hyper_3_5_chunk | new_space
+ _hyper_3_6_chunk | new_space
+ _hyper_3_7_chunk | new_space
+ _hyper_3_8_chunk | new_space
+ _hyper_3_9_chunk | new_space
+ hyper_in_space   | new_space
+(9 rows)
+
+\set ON_ERROR_STOP 0
+-- not an empty tablespace
+DROP TABLESPACE new_space;
+ERROR:  tablespace "new_space" is not empty
+\set ON_ERROR_STOP 1
+SELECT drop_chunks(20, 'hyper_in_space');
+ drop_chunks 
+-------------
+ 
+(1 row)
+
+SELECT tablename, tablespace FROM pg_tables WHERE tablespace = 'new_space' ORDER BY tablename;
+   tablename    | tablespace 
+----------------+------------
+ hyper_in_space | new_space
+(1 row)
+
+\set ON_ERROR_STOP 0
+-- should not be able to drop tablespace if a hypertable depends on it
+-- even when there are no chunks
+DROP TABLESPACE new_space;
+ERROR:  tablespace "new_space" is not empty
+\set ON_ERROR_STOP 1
+DROP TABLE hyper_in_space;
+CREATE TABLE hyper_in_space(time bigint, temp float, device int) TABLESPACE new_space;
+SELECT create_hypertable('hyper_in_space', 'time', 'device', 4, chunk_time_interval=>1);
+NOTICE:  adding not-null constraint to column "time"
+ create_hypertable 
+-------------------
+ 
+(1 row)
+
+INSERT INTO hyper_in_space(time, temp, device) VALUES (1, 20, 1);
+INSERT INTO hyper_in_space(time, temp, device) VALUES (3, 21, 2);
+INSERT INTO hyper_in_space(time, temp, device) VALUES (5, 23, 1);
+SELECT tablename, tablespace FROM pg_tables
+WHERE tablename = 'hyper_in_space' OR tablename ~ '_hyper_\d+_\d+_chunk' ORDER BY tablename;
+     tablename     | tablespace 
+-------------------+------------
+ _hyper_1_1_chunk  | 
```

Comment:
Should drop the old tables before running this test, it's hard to see what's from the recent inserts and what's old.

Reply:
dropping is not ideal I think. will try to constrain to this table only by sth like this
```
SELECT CONCAT(c.schema_name, '.',c.table_name)::REGCLASS AS chunk
        FROM _timescaledb_catalog.chunk c
        INNER JOIN _timescaledb_catalog.hypertable h ON (h.id = c.hypertable_id)
        AND h.schema_name = _schema_name
        AND h.table_name = _table_name;
```
Or if `show_chunks` PR merges before this, I can just use show chunks.

Reply:
Will that show if we're creating spurious tables?

Reply:
would not, you are right.
went with deleting tables approach